js/utils.js

```diff
@@ -83,16 +81,19 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
         }
 
         var filesDetailViewInfo = Promise.all(files.map(function(fileUrl) {
-          var fileId = Fliplet.Media.isRemoteUrl(fileUrl)[2];
+          var fileId = fileUrl.match(/v1\/media\/files\/([0-9]+)/)[1];
 
           return Fliplet.Media.Files.get(fileId).then(function(file) {
             return {
               name: file.name,
-              size: file.size,
+              size: file.metadata.size,
               uploaded: file.createdAt,
               url: file.url
             };
-          });
+          })
+            .catch(function(err) {
```

Comment:
We can reject the error without parsing it. The parsing should only happen right before something needs to be done with the error, e.g. before displaying it. This way the error object can be passed to the callback function in its full form for assessment.

```suggestion
            .catch(reject);
```